microcosm_eventsource/func.py

```diff
@@ -0,0 +1,67 @@
+"""
+Custom database functions.
+
+"""
+from sqlalchemy.sql.expression import FunctionElement
+from sqlalchemy.ext.compiler import compiles
+
+
+class last(FunctionElement):
```

Comment:
Extending `FunctionElement` is how you define custom function in `SQLAlchemy`. In turn, we can do things like:

```
last(Foo.bar).over(...)
```